### DISCUSSION TITLE
Publish to azure: The endpoint blob is not allocated for the resource storage

### DISCUSSION THREAD

**Discussion by BjarkeKvik** [microsoft/aspire#5882] [Q&A] [answered]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the bug

# Issue Description

### The project runs and API works as expected when running locally.
# Code
### Program.cs (host)
```
using Microsoft.Extensions.Hosting;
using Projects;

var builder = DistributedApplication.CreateBuilder(args);
var storage = builder.AddAzureStorage("storage");
var blobs = storage.AddBlobs("blobs");

if (builder.Environment.IsDevelopment())
{
    storage.RunAsEmulator()
    .WithAnnotation(new ContainerImageAnnotation
    {
        Registry = "mcr.microsoft.com",
        Image = "azure-storage/azurite",
        Tag = "3.31.0"
    });
}
builder.AddProject<BlobStorage_Api>("blobstorageapi").WithReference(blobs);
builder.Build().Run();
```
### Program.cs (api)
```
using BlobStorage.Foundation.Interfaces;
using BlobStorage.Implementation;

var builder = WebApplication.CreateBuilder(args);

builder.Services.AddControllers();
builder.Services.AddEndpointsApiExplorer();
builder.Services.AddSwaggerGen();

builder.Services.AddScoped<IBlobLibrary, BlobRepository>();
builder.Services.AddScoped<BlobService>();

builder.AddAzureBlobClient("blobs");

var app = builder.Build();

if (app.Environment.IsDevelopment())
{
    app.UseSwagger();
    app.UseSwaggerUI();
}
app.UseHttpsRedirection();

app.UseAuthorization();

app.MapControllers();

app.Run();
```

### Expected Behavior

I expect to be able to deploy my Azure Blob Storage to Azure.

### Exceptions (if any)

# Details:
•	Command: azd init
•	Project: C:\Users\user\source\repos\BlobStorageApi\Aspire\Aspire.AppHost\Aspire.AppHost.csproj
•	Exit Code: 3762504530
•	Output:
•	Building...
•	Info: Aspire version: 8.0.0+d215c528c07c7919c3ac30b35d92f4e51a60523b
•	Error: Hosting failed to start
•	Exception: System.InvalidOperationException:   
 ### The endpoint 'blob' is not allocated for the resource 'storage'.
•	Stack Trace:
•	Aspire.Hosting.ApplicationModel.EndpointReference.get_AllocatedEndpoint()
•	Aspire.Hosting.ApplicationModel.EndpointReference.get_Port()
•	Aspire.Hosting.Azure.AzureStorageResource.GetBlobConnectionString()
•	Aspire.Hosting.Azure.AzureBlobStorageResource.get_ConnectionStringExpression()
•	Aspire.Hosting.Publishing.ManifestPublishingContext.WriteConnectionString(IResource resource)
•	Aspire.Hosting.Publishing.ManifestPublishingContext.WriteConnectionStringAsync(IResourceWithConnectionString resource)
•	Aspire.Hosting.Publishing.ManifestPublishingContext.<>c__DisplayClass20_4.<WriteResourceAsync>b__5()
•	Aspire.Hosting.Publishing.ManifestPublishingContext.<>c__DisplayClass20_0.

### .NET Version info

[aspire]
   Installation Source: SDK 8.0.300, VS 17.11.35005.142
   Manifest Version:    8.0.0/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.aspire\8.0.0\WorkloadManifest.json
   Install Type:        FileBased

**Comment by davidfowl** [comment, marked as answer]
Change your code to be this:

```C#
using Microsoft.Extensions.Hosting;
using Projects;

var builder = DistributedApplication.CreateBuilder(args);
var storage = builder.AddAzureStorage("storage");
var blobs = storage.AddBlobs("blobs");

if (builder.Environment.IsDevelopment())
{
    storage.RunAsEmulator(c => c.WithImageTag("3.31.0"));
}
builder.AddProject<BlobStorage_Api>("blobstorageapi").WithReference(blobs);
builder.Build().Run();
```

**Comment by BjarkeKvik** [comment]
It worked! Thank you
